File: backend/app/schemas/analytics.py
from datetime import datetime
from typing import List

from pydantic import BaseModel


class ScoreTrendPoint(BaseModel):
    date: datetime
    average_score: float
    submissions: int


class KeywordFrequency(BaseModel):
    keyword: str
    count: int


class AnalyticsSummary(BaseModel):
    total_resumes: int
    average_score: float
    keywords: List[KeywordFrequency]
    score_trends: List[ScoreTrendPoint]

File: frontend/src/types/analytics.ts
export type KeywordFrequency = {
  keyword: string;
  count: number;
};

export type ScoreTrendPoint = {
  date: string;
  average_score: number;
  submissions: number;
};

export type AnalyticsSummary = {
  total_resumes: number;
  average_score: number;
  keywords: KeywordFrequency[];
  score_trends: ScoreTrendPoint[];
};
